lib/deaf_grandma.rb

```diff
@@ -20,12 +20,34 @@ def run!
 
   def speak(input)
 
-    #Implement your code here <<<<<<<<<
-
+    if is_yelling(input)
+        "NOT SINCE 1964!"
+    else
+        "SPEAK UP SONNY!"
+    end
   end
 
   private
 
+    def is_yelling(text)
+        return false if text != text.upcase
+
+        if text == "BYE"
+            @bye_counter += 1
```

Comment:
I think it is a bit weird to mutate `@bye_counter` while determining whether the input is "yelling" or not. Perhaps you can extract another method somehow?